code/datums/outputs.dm

```diff
@@ -0,0 +1,117 @@
+//NOTE: When adding new sounds here, check to make sure they haven't been added already via CTRL + F.
+
+/datum/outputs
+    var/text = "You hear something."
+    var/list/sounds = 'sound/misc/bangindonk.ogg' //can be either a sound path or a WEIGHTED list, put multiple for random selection between sounds
+    // var/mutable_appearance/vfx = list('icons/sound_icon.dmi',"circle", HUD_LAYER) //syntax: icon, icon_state, layer
+    var/mutable_appearance/vfx = list('icons/mob/talk.dmi',"scream", HUD_LAYER) //syntax: icon, icon_state, layer
+    var/cooldown =  0.1 SECONDS // ms
+
+/datum/outputs/New()
+    vfx = mutable_appearance(vfx[1],vfx[2],vfx[3])
+
+/datum/outputs/proc/send_info(mob/receiver, turf/turf_source, vol as num, vary, frequency, falloff, channel = 0, pressure_affected = TRUE, last_played_time)
+    var/sound/sound_output
+    //Pick sound
+    if(islist(sounds))
+        if(sounds.len)
+            var/soundin = pickweight(sounds)
+            sound_output = sound(get_sfx(soundin))
+    else
+        sound_output = sound(get_sfx(sounds))
+    //Process sound
+    if(sound_output)
+        sound_output.wait = 0 //No queue
+        sound_output.channel = channel || 0 // TODO: open_sound_channel()
+        sound_output.volume = vol
+
+        if(vary)
+            if(frequency)
+                sound_output.frequency = frequency
+            else
+                sound_output.frequency = GET_RANDOM_FREQ
+
+        if(isturf(turf_source))
+            var/turf/T = get_turf(receiver)
+
+            //sound volume falloff with distance
+            var/distance = get_dist(T, turf_source)
+
+            sound_output.volume -= max(distance - world.view, 0) * 2 //multiplicative falloff to add on top of natural audio falloff.
+
+            if(pressure_affected)
+                //Atmosphere affects sound
+                var/pressure_factor = 1
+                var/hearer_pressure = T.return_pressure()
+                var/source_pressure = turf_source.return_pressure()
+                // var/datum/gas_mixture/hearer_env = T.return_air()
```

Comment:
Same here

Reply:
meant to have been removed, i missed this